### DISCUSSION TITLE
Can e3nn use 2nd order tensor features (instead of a vector) ?

### DISCUSSION THREAD

**Discussion by jiachenguoNU** [e3nn/e3nn#376] [Q&A]
Hi,

I'm reading the Nature Communication paper (on NequIP) and checking the code for e3nn. In the paper, it is pointed out that the features on the atom can be tensors of different orders such as scalars, vectors and higher-order tensors. I can see from the example that scalars and vectors can be used as features (I guess that's because spherical harmonics are equivariant and we can project a vector function into the spherical harmonics.)

But I don't know if I can use some other higher-order tensors such as the 2nd order symmetric tensor (stress/strain) as the feature on the atoms. Could you please give some insights on this issue?

**Comment by mariogeiger** [comment]
Any (finite dimensional) representation can be decomposed into (a finite number of) irreducible representations. 

For instance a 3x3 symmetric tensor decompose into a scalar and a irrep l=2. 

The appropriate change of basis can be obtained using the function https://docs.e3nn.org/en/stable/api/o3/o3_reduce.html
